project/npda/models/auditcohort.py

```diff
@@ -0,0 +1,108 @@
+from datetime import date
+from typing import Any, Iterable
+from django.db import models
+
+
+class AuditCohort(models.Model):
+    """
+    The AuditCohort class.
+
+    This class is used to define the cohort of patients that are being audited. The cohort tracks the progress of the audit
+    """
+
+    audit_year = models.IntegerField(
+        "Audit year",
+        blank=False,
+        null=False,
+        help_text="Year the audit is being conducted",
+    )
+
+    cohort_number = models.IntegerField(
+        "Cohort number",
+        blank=False,
+        null=False,
+        help_text="The cohort number of the patient",
+    )
+
+    pz_code = models.CharField(
+        "PZ code",
+        max_length=10,
+        blank=False,
+        null=False,
+        help_text="The PZ code of the Paediatric Diabetes Unit",
+    )
+
+    ods_code = models.CharField(
+        "PZ code",
+        max_length=10,
+        blank=False,
+        null=False,
+        help_text="The ODS code of the Organisation",
+    )
+
+    submission_date = models.DateTimeField(
+        "Submission date",
+        help_text="Date the submission was created",
+    )
+
+    submission_active = models.BooleanField(
+        "Submission active",
+        default=True,
+        help_text="Submission is active and being considered for inclusion in the audit",
+    )
+
+    submission_by = models.ForeignKey(
+        on_delete=models.CASCADE,
+        to="npda.NPDAUser",
+    )
+
+    submission_approved = models.BooleanField(
+        "Submission approved",
+        default=False,
+        help_text="Submission has been approved for inclusion in the audit",
+    )
+
+    patients = models.ManyToManyField(to="npda.Patient", related_name="audit_cohorts")
+
+    def calculate_cohort_number(self):
+        """
+        Returns the cohort number of the patient
+
+        **The audit year starts on the 1st of April and ends on the 31st of March the following year**
+        Returns 1 if the patient has less than 25% of the audit year remaining
+        Returns 2 if the patient has less than 50% of the audit year remaining
+        Returns 3 if the patient has less than 75% of the audit year remaining
+        Returns 4 if the patient has more than 75% of the audit year remaining
+        """
+        audit_start_date = date(self.audit_year, 4, 1)
+        if self.submission_date.date() < audit_start_date:
+            # The patient was audited in the previous year
+            days_remaining = (audit_start_date - self.submission_date.date()).days
+        else:
+            # The patient was audited in the current year
+            audit_end_date = date(self.audit_year + 1, 3, 31)
+            days_remaining = (audit_start_date - self.submission_date.date()).days
```

Comment:
Only potential logic error I can see might be here, and on the same code above in the if statement (line 80). If we're calculating days remaining in this current cohort shouldn't we be using `audit_end_date` instead of `audit_start_date`?

Reply:
This has been actioned in my most recent commits to this PR. Will wait for confirmation from @eatyourpeas then happy to resolve this conversation.